ContainerHandling/New-NavImage.ps1

```diff
@@ -104,300 +105,306 @@ function New-BcImage {
         $hostOs = "ltsc2016"
     }
 
-    if (!$imageName.Contains(':')) {
-        $appUri = [Uri]::new($artifactUrl)
-        $imageName += ":$($appUri.AbsolutePath.ToLowerInvariant().Replace('/','-').TrimStart('-'))"
-        if ($skipDatabase) {
-            $imageName += "-nodb"
-        }
-        if ($multitenant) {
-            $imageName += "-mt"
-        }
-    }
-
-    Write-Host "Building image $imageName based on $baseImage"
-    
-    $imageName
-
-    if ($baseImage -eq $bestGenericImageName) {
-        Write-Host "Pulling latest image $baseImage"
-        DockerDo -command pull -imageName $baseImage | Out-Null
+    if (!$imageName.Contains(':')) {
+        $appUri = [Uri]::new($artifactUrl)
+        $imageName += ":$($appUri.AbsolutePath.ToLowerInvariant().Replace('/','-').TrimStart('-'))"
+        if ($skipDatabase) {
+            $imageName += "-nodb"
+        }
+        if ($multitenant) {
+            $imageName += "-mt"
+        }
     }
-    else {
-        $baseImageExists = docker images --format "{{.Repository}}:{{.Tag}}" | Where-Object { $_ -eq "$baseImage" }
-        if (!($baseImageExists)) {
-            Write-Host "Pulling non-existing base image $baseImage"
+
+    $exists = (DockerDo -command images -imageName $imageName | Out-Null)

```

Comment:
This is an actual change. It checks if an image with the name that would be used for the build already exists. If it exists and the new switch is yes, then it skips the rest